ONSdigital/eq-author-app
# Add a script to grant firebase users admin roles.

### What is the context of this PR?
There are instances where it would be useful to restrict certain features of Author to only "admins" within the team. An example of this might be restricting access import/export functionality in different environments, being able to manage user accounts, A/B testing new features before they are released to all users etc.

Whilst the permissions model is intended to address the issue of fine grained access to questionnaires, being able to give certain users admin privileges feels like an admin process that should happen outside of the application itself.

This PR introduces an admin script that can be run by a firebase service account to grant (and remove) admin status for any firebase user within the firebase project.

### How to review 
- Create a test firebase project.
- Enable authentication.
- Create a user within the firebase project.
- Create a service account for the firebase project and download the JSON key file.
- Run the script using the project name, uid and service account key.
- If successful the user should have the custom claim { admin: true } in their JWT token the next time they log in.
- Run the script again with the `-r` flag to remove the admin privilege.
